apps/chat/src/utils/app/data/storages/api/application-api-storage.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  getApplicationApiKey,
+  parseApplicationApiKey,
+} from '@/src/utils/server/api';
+
+import {
+  ApplicationInfo,
+  CustomApplicationModel,
+} from '@/src/types/applications';
+import { ApiKeys, Entity } from '@/src/types/common';
+
+import {
+  ApiApplicationModel,
+  ApplicationDetailsResponse,
+  convertApplicationFromApi,
+  convertApplicationToApi,
+} from '../../../application';
+import { ApiEntityStorage } from './api-entity-storage';
+
+export class ApplicationApiStorage extends ApiEntityStorage<
+  ApplicationInfo,
+  CustomApplicationModel,
+  ApplicationDetailsResponse,
+  ApiApplicationModel
+> {
+  mergeGetResult(
+    info: Entity,
+    entity: ApplicationDetailsResponse,
+  ): CustomApplicationModel {
+    return {
+      ...info,
+      ...convertApplicationFromApi(entity),
+    };
+  }
+  cleanUpEntity(application: CustomApplicationModel): ApiApplicationModel {
+    return convertApplicationToApi(application);
+  }
+  getEntityKey(info: ApplicationInfo): string {
+    return getApplicationApiKey(info);
+  }
+  parseEntityKey(key: string): Omit<ApplicationInfo, 'folderId' | 'id'> {
+    return parseApplicationApiKey(key);
+  }
+  getStorageKey(): ApiKeys {
```

Comment:
You don't need to define `ApiKeys` return type here. It automatically will be `ApiKeys.Applications`. `ApiKeys` may be incorrect here

Reply:
What do you mean?